### ISSUE TITLE
Error Message in Weewx Logs after activating Belchertown Skin

### ISSUE THREAD

**Issue by ebnerjoh** [poblabs/weewx-belchertown#49]
Hi,

Since I have activated the Belchertown Skin I can see the following error message whenever the archive is executed (5 minutes)

```
Jan 20 08:30:47 weewx weewx[3035]: cheetahgenerator: Generate failed with exception '<type 'exceptions.ValueError'>'
Jan 20 08:30:47 weewx weewx[3035]: cheetahgenerator: **** Ignoring template /etc/weewx/skins/Belchertown/index.html.tmpl
Jan 20 08:30:47 weewx weewx[3035]: cheetahgenerator: **** Reason: invalid literal for float(): 0,6
Jan 20 08:30:47 weewx weewx[3035]: ****  Traceback (most recent call last):
Jan 20 08:30:47 weewx weewx[3035]: ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 330, in generate
Jan 20 08:30:47 weewx weewx[3035]: ****      print >> _file, compiled_template
Jan 20 08:30:47 weewx weewx[3035]: ****    File "/usr/lib/python2.7/dist-packages/Cheetah/Template.py", line 1005, in __str__
Jan 20 08:30:47 weewx weewx[3035]: ****      rc = getattr(self, mainMethName)()
Jan 20 08:30:47 weewx weewx[3035]: ****    File "_etc_weewx_skins_Belchertown_index_html_tmpl.py", line 1572, in respond
Jan 20 08:30:47 weewx weewx[3035]: ****  ValueError: invalid literal for float(): 0,6
```

I do not see any issues (except that I am still not able to get mqtt running with the skin, but I am sure this is not related).

Br,
Johannes

**Comment by poblabs** [comment]
I think you're missing some of the error. Paste everything from where it says "Belchertown Skin Version: XX". 

If you don't see that, then you have an old version and please [upgrade to the latest version](https://github.com/poblabs/weewx-belchertown/releases).

**Comment by ebnerjoh** [comment]
I think it is resolved. I had the latest version installed, but I changed the locales of raspbian to de. Changed back to the original one and now the issue is gone!

Br,
Johannes

**Comment by poblabs** [comment]
Ok sounds good. I made a change about 4-5 days ago that fixes locales. If you want to test it out that'd be great. 

Download the [latest development belchertown.py ](https://raw.githubusercontent.com/poblabs/weewx-belchertown/master/bin/user/belchertown.py)and place it into your bin/user folder. Then restart weewx. If no errors, change your locale again and restart. Then see how it goes. Let me know.

**Comment by ebnerjoh** [comment]
Belchertown Extension: version 0.9rc1

**Comment by poblabs** [comment]
@ebnerjoh I'm trying to replicate this and I can't. Here's my settings. 

Running the beta test skin version 0.9rc1. 

Using de_DE.UTF-8 as my system locale:

```
root@weewxdeutf:~$ locale
LANG=de_DE.UTF-8
LANGUAGE=
LC_CTYPE="de_DE.UTF-8"
LC_NUMERIC="de_DE.UTF-8"
LC_TIME="de_DE.UTF-8"
LC_COLLATE="de_DE.UTF-8"
LC_MONETARY="de_DE.UTF-8"
LC_MESSAGES="de_DE.UTF-8"
LC_PAPER="de_DE.UTF-8"
LC_NAME="de_DE.UTF-8"
LC_ADDRESS="de_DE.UTF-8"
LC_TELEPHONE="de_DE.UTF-8"
LC_MEASUREMENT="de_DE.UTF-8"
LC_IDENTIFICATION="de_DE.UTF-8"
LC_ALL=
```

Skin generates reports without error. 

```
Jan 30 16:00:16 weewxdeutf weewx[936]: cheetahgenerator: Generated 4 files for report Highcharts_Belchertown in 0.69 seconds
Jan 30 16:00:16 weewxdeutf weewx[936]: copygenerator: copied 0 files to /var/www/html/weewx/belchertown
Jan 30 16:00:16 weewxdeutf weewx[936]: Belchertown Extension: version 0.9rc1
Jan 30 16:00:16 weewxdeutf weewx[936]: cheetahgenerator: Generated 14 files for report Belchertown in 0.74 seconds
Jan 30 16:00:16 weewxdeutf weewx[936]: copygenerator: copied 29 files to /var/www/html/weewx/belchertown
```

2 questions: 

1. Are you sure you're running 0.9rc1 and not 0.8.2?
2. Can you paste the output of `locales`?

Thanks

**Comment by ebnerjoh** [comment]
Hi, 

Regarding your 1st question: If I am running wee_extension --list it shows me v0.8.2 but I have downloaded the latest files from github.

**Comment by poblabs** [comment]
Ok. I'm still not able to replicate this. Please send the output of locales. Thanks.

**Comment by poblabs** [comment]
Strange. Still can't replicate it. I'll keep trying. 

`Feb  2 15:50:50 weewxdeutf weewx[888]: engine: Locale is 'de_DE.UTF-8'`


```
Feb  2 15:55:17 weewxdeutf weewx[938]: cheetahgenerator: Generated 4 files for report Highcharts_Belchertown in 0.55 seconds
Feb  2 15:55:17 weewxdeutf weewx[938]: copygenerator: copied 0 files to /var/www/html/weewx/belchertown
Feb  2 15:55:17 weewxdeutf weewx[938]: Belchertown Extension: version 0.9rc1
Feb  2 15:55:18 weewxdeutf weewx[938]: cheetahgenerator: Generated 14 files for report Belchertown in 0.70 seconds
Feb  2 15:55:18 weewxdeutf weewx[938]: copygenerator: copied 29 files to /var/www/html/weewx/belchertown
```

**Comment by poblabs** [comment]
Finally was able to replicate this. Now to break it down and find out what's going on.

```
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: Belchertown Extension: version 0.9rc4
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: cheetahgenerator: Generate failed with exception '<type 'exceptions.ValueError'>'
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: cheetahgenerator: **** Ignoring template /etc/weewx/skins/Belchertown/index.html.tmpl
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: cheetahgenerator: **** Reason: invalid literal for float(): 0,0
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****  Traceback (most recent call last):
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 330, in generate
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****      print >> _file, compiled_template
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****    File "/usr/lib/python2.7/dist-packages/Cheetah/Template.py", line 1005, in __str__
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****      rc = getattr(self, mainMethName)()
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****    File "_etc_weewx_skins_Belchertown_index_html_tmpl.py", line 1547, in respond
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****  ValueError: invalid literal for float(): 0,0
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: cheetahgenerator: Generate failed with exception '<type 'exceptions.ValueError'>'
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: cheetahgenerator: **** Ignoring template /etc/weewx/skins/Belchertown/pi/index.html.tmpl
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: cheetahgenerator: **** Reason: invalid literal for float(): 0,0
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****  Traceback (most recent call last):
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 330, in generate
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****      print >> _file, compiled_template
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****    File "/usr/lib/python2.7/dist-packages/Cheetah/Template.py", line 1005, in __str__
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****      rc = getattr(self, mainMethName)()
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****    File "_etc_weewx_skins_Belchertown_pi_index_html_tmpl.py", line 1199, in respond
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: ****  ValueError: invalid literal for float(): 0,0
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: cheetahgenerator: Generated 12 files for report Belchertown in 0.71 seconds
Feb  6 02:35:52 weewxdeutf wee_reports[21148]: copygenerator: copied 30 files to /var/www/html/weewx/belchertown
```

**Comment by poblabs** [comment]
This error was generated from a `float()` and the windSpeed having a comma in it. Like `0,0`. This is fixed, but will require weewx 3.9.0 and Belchertown skin 0.9. 

weewx 3.9.0 is out now. 

Belchertown skin 0.9 needs another day or two. 

Feel free to grab the beta. Download the master zip from the front page and manually copy all the files to their locations (`skins/Belchertown`, `skins/Highcharts_Belchertown` and `bin/user`). Then restart weewx. 

**Have a backup of your files and database just in case.**

**Comment by ebnerjoh** [comment]
Hi,

Yes, it is working now!

Br,
Johannes

Am Mi., 6. Feb. 2019 um 03:58 Uhr schrieb poblabs <notifications@github.com
